packages/discovery-react-components/src/components/ResultsPagination/ResultsPagination.tsx

```diff
@@ -126,17 +159,19 @@ const ResultsPagination: FC<ResultsPaginationProps> = ({
       <>
         {!isResultsPaginationComponentHidden && (
           <CarbonPagination
-            className={classNames.join(' ')}
-            initialPage={currentPage}
+            className={cx(`${settings.prefix}--pagination`, {
+              [`${settings.prefix}--pagination__page-size-selector--hidden`]: !showPageSizeSelector
+            })}
             totalItems={matchingResults}
-            pageSize={actualPageSize}
+            pageSize={resultsPerPage}
             pageSizes={pageSizes}
             // NOTE: See PageSelector below for details about `onChange`
             onChange={handleOnChange}
             itemRangeText={handleItemRangeText}
             itemsPerPageText={mergedMessages.itemsPerPageText}
             pageRangeText={handlePageRangeText}
             pageText={handlePageText}
+            key={resetCounter}
```

Comment:
Can you add a comment here or elsewhere explaining why you use a counter here?

Reply:
I added a comment within the reset function.